encode.go

```diff
@@ -92,34 +110,33 @@ func Encode(dst, src []byte) []byte {
 		if len(src) != 0 {
 			d += emitLiteral(dst[d:], src)
 		}
+		e.cur += 4
 		return dst[:d]
 	}
 
-	// Initialize the hash table. Its size ranges from 1<<8 to 1<<14 inclusive.
-	const maxTableSize = 1 << 14
-	shift, tableSize := uint(32-8), 1<<8
-	for tableSize < maxTableSize && tableSize < len(src) {
```

Comment:
There is no benefit of keeping it. Less bits will lead to worse matches, which will lead to more tests (ie it is slower) and less compression.

IMO you can add the "compatibility mode" later, and if it doesn't impact speed I see no problem.